### PR TITLE
MODGOBI-77 Contributor-name-type is required for contributor added to POL

### PR DESCRIPTION
**Pull request by piotr-kalashuk**
## Purpose
[MODGOBI-77](https://issues.folio.org/browse/MODGOBI-77) Contributor-name-type is required for contributor added to POL
## Approach
In case the contributor is resolved, the logic searches for contributor name type id in the inventory by `Personal name` name or any if there is no such. If no any contributor name type found, the logic logs an error and ignores contributor (contributor name type is required field and without it validation will reject entire order)

#### TODOS and Open Questions
- [ ] Point to acq-models master commit once https://github.com/folio-org/acq-models/pull/145 is merged
- [ ] Merge together with changes for [MODORDERS-247](https://issues.folio.org/browse/MODORDERS-247): https://github.com/folio-org/mod-orders/pull/182

## Pre-Merge Checklist
Before merging this PR, please go through the following list and take appropriate actions.

- Does this PR meet or exceed the expected quality standards?
  - [x] Code coverage on new code is 80% or greater
  - [x] Duplications on new code is 3% or less
  - [x] There are no major code smells or security issues
- Does this introduce breaking changes?
  - [ ] Were any API paths or methods changed, added or removed?
  - [ ] Were there any schema changes?
  - [ ] Did any of the interface versions change?
  - [ ] Were permissions changed, added, or removed?
  - [ ] Are there new interface dependencies?
  - [ ] There are no breaking changes in this PR.

### CODE REVIEW

File: src/main/java/org/folio/gobi/Mapper.java [hunk 1]
```diff
@@ -712,19 +731,7 @@ private void mapVendorDetail(List<CompletableFuture<?>> futures, VendorDetail ve
...
-        if (f.get(instance) != null)
-          return false;
-      } catch (IllegalArgumentException e) {
-        logger.error("Unable to determine Object", e);
-      } catch (IllegalAccessException e) {
-        logger.error("Unable to access Object", e);
-      }
-    }
-    return true;
+    return JsonObject.mapFrom(instance).isEmpty();
```
Comment (craigmcnally) [inline src/main/java/org/folio/gobi/Mapper.java]: much cleaner!

File: src/main/java/org/folio/rest/impl/PostGobiOrdersHelper.java [hunk 2]
```diff
@@ -307,6 +310,34 @@ public PostGobiOrdersHelper(HttpClientInterface httpClient,
...
+   */
+  public CompletableFuture<String> lookupContributorNameTypeId(String name) {
+    logger.info("Received contributorNameType is {}", name);
+    String query = HelperUtils.encodeValue(String.format(CQL_NAME_CRITERIA, name), logger);
+    String endpoint = String.format(CONTRIBUTOR_NAME_TYPES_ENDPOINT + QUERY + LIMIT_1, query);
+    return handleGetRequest(endpoint).thenApply(HelperUtils::extractContributorNameTypeId)
+      .thenCompose(typeId -> {
+        if (StringUtils.isEmpty(typeId)) {
+          if (DEFAULT_LOOKUP_CODE.equals(name)) {
+            logger.error("No any contributorNameType available");
```
Comment (craigmcnally) [inline src/main/java/org/folio/rest/impl/PostGobiOrdersHelper.java]: this is petty but please correct this to "No contributorNameTypes are available"

File: src/test/resources/MockData/compositePurchaseOrder.json [hunk 3]
```diff
@@ -54,6 +55,7 @@
       "materialType" : "be44c321-ab73-43c4-a114-70f82fa13f17"
     },
     "fundDistribution" : [ {
+      "fundId" : "f5f1c495-7f7a-477a-b2e9-10eb043e37f4",
       "code" : "f5f1c495-7f7a-477a-b2e9-10eb043e37f4",
```
Comment (craigmcnally) [inline src/test/resources/MockData/compositePurchaseOrder.json]: Not really important but fundDistribution.code should be a short, human readable alphabetical code, e.g. USHIST

File: src/main/java/org/folio/rest/impl/PostGobiOrdersHelper.java (hunk 2 above)
Comment (piotr-kalashuk) [reply src/main/java/org/folio/rest/impl/PostGobiOrdersHelper.java]: Corrected

File: src/test/resources/MockData/compositePurchaseOrder.json (hunk 3 above)
Comment (piotr-kalashuk) [reply src/test/resources/MockData/compositePurchaseOrder.json]: Updated

File: src/test/java/org/folio/rest/impl/GOBIIntegrationServiceResourceImplTest.java [hunk 4]
```diff
@@ -1398,6 +1215,31 @@ private void handleGetProductTypes(RoutingContext ctx) {
         .end(productTypes.encodePrettily());
     }
 
+    private void handleGetContributorNameTypes(RoutingContext ctx) {
+      logger.info("got contributorNameTypes request: {}", ctx.request().query());
+      String instruction = ctx.request().getHeader(MOCK_OKAPI_GET_CONTRIBUTOR_NAME_HEADER);
+
+      JsonObject productTypes = new JsonObject();
```
Comment (pautau-s) [inline src/test/java/org/folio/rest/impl/GOBIIntegrationServiceResourceImplTest.java]: I suppose this variable should be renamed `сontributorNameTypes`

Comment (piotr-kalashuk) [reply src/test/java/org/folio/rest/impl/GOBIIntegrationServiceResourceImplTest.java]: Corrected

**Pull request merged**